src/components/select-project/select-project.controller.ts

```diff
@@ -277,4 +288,10 @@ export class SelectProjectController implements angular.IController {
       });
     }
   }
+
+  private getProjectByName() {
+    return _.find(this.projects, (project: any) => {
```

Comment:
You could use the shorthand for `_.find` here.

Seems strange to have a method called `getProjectByName` that doesn't take a name as a parameter. I'd probably make it general and take in the name.

```ts
private getProjectByName(name) {
  return _.find(this.projects, { metadata: { name: name } });
}
```

Reply:
I know, but it's already on the $scope, but I understand.  I tried to use the shorthand, with a nested obj, didn't work, so I went long hand.

Reply:
> I tried to use the shorthand, with a nested obj, didn't work, so I went long hand.

It should work: we do it in a lot of places. Can you paste the code?

Reply:
I was thinking something like `preselectProject()` would be a better abstraction with lines 253-259. Something like

```ts
private preselectProject() {
  if (this.ctrl.selectedProject) {
    return;
  }

  // if one project, default to it, else default to selected project by name (if available)
  if (_.size(this.projects) === 1) {
    this.ctrl.selectedProject = this.projects[0];
  } else if (this.ctrl.selectedProjectName) {
    this.ctrl.selectedProject = _.find(this.projects, {
      metadata: {
        name: this.ctrl.selectedProjectName
      }
    });
  }
  this.onSelectProjectChange();
}
```